ensembler/potentials/biasOneD.py

```diff
@@ -112,12 +104,365 @@ def return_path_correction_term(self):
+    position = sp.symbols("r")
+
+    def __init__(self, origPotential, amplitude=0.1, sigma=0.1, n_trigger=100, bias_grid_min=0, bias_grid_max=10, numbins=100  ):
+
+        '''
+        This is the Constructor of the metadynamicsPotential class.
+        Parameters
+        ----------
+        origPotential: potential type
+            The unbiased potential
+        amplitude : float
+            scaling of the gaussian potential added in the metadynamcis step
+        sigma: float
+            standard deviation of the gaussian potential added in the metadynamcis step
+        n_trigger : int
+            Metadynamics potential will be added after every n_trigger'th steps
+        bias_grid_min: float
+            min value of the bias grid
+        bias_grid_max: float
+            max value of the bias grid
+        numbins: float
+            size of the grid bias and forces are saved in
+        '''
+
+        self.origPotential = origPotential
+        self.n_trigger = n_trigger
+        self.amplitude = amplitude
+        self.sigma = sigma
+        #grid where the bias is stored
+        #currently only for 1D
+        self.bias_grid_energy = np.zeros(numbins)   # energy grid
+        self.bias_grid_force = np.zeros(numbins)  # force grid
+        # get center value for each bin
+        bin_half = (bias_grid_max - bias_grid_min)/(2*numbins) # half bin width
+        self.bin_centers = np.linspace(bias_grid_min+bin_half,bias_grid_max-bin_half, numbins)
+        #current_n counts when next metadynamic step should be applied
+        self.current_n = 1
+        # count how often the potential was updated
+        self.finished_steps = 0
+
+        self.constants = {**origPotential.constants, **gaussPot.constants}
+
+        self.V_orig = origPotential.V
+        self.V_orig_part = self.V_orig.subs(self.constants)
+        self.dVdpos = sp.diff(self.V_orig_part, self.position)
+        self.V = self.V_orig_part
+
+        super().__init__()
+
+    def check_for_metastep(self, curr_position):
+        '''
+        Checks if the bias potential should be added at the current step
+        Parameters
+        ----------
+        curr_position: float
+            current x position
+
+        Returns
+        -------
+        '''
+
+        if self.current_n%self.n_trigger == 0:
+            self._update_potential(curr_position)
+            self.finished_steps += 1
+            self.current_n = 1
+        else:
+            self.current_n += 1
+
+    def _update_potential(self, curr_position):
+        '''
+        Is triggered by check_for_metastep(). Adds a gaussian centered on the
+        current position to the potential
+
+        Parameters
+        ----------
+        curr_position: float
+            current x position
+
+        Returns
+        -------
+        '''
+        # do gaussian metadynamics
+        new_bias = gaussPot(A=self.amplitude, mu=curr_position, sigma=self.sigma)
```

Comment:
could be a class attribute, allowing different biases (self.bias - initialized on construction (with a defautl on gaussPot) and update the mu here)